src/script/common/c_content.cpp

```diff
@@ -66,8 +66,7 @@ ItemDefinition read_item_definition(lua_State* L,int index,
 	lua_pop(L, 1);
 
 	def.stack_max = getintfield_default(L, index, "stack_max", def.stack_max);
-	if(def.stack_max == 0)
-		def.stack_max = 1;
+	def.stack_max = rangelim(def.stack_max, 1, (u16)-1);
```

Comment:
The latter condition will always be true because of the range of stack_max's type.  If the user enters a number greater than U16_MAX, there will be a silent overflow and the effective stack size will be the modder's intended stack size modulo 65536, instead of being truncated at 65535 like you want.
You need to fix getintfield() to solve this the proper way.